beets/importer.py

```diff
@@ -546,10 +546,22 @@ def _emit_imported(self, lib):
             return
         plugins.send('album_imported', lib=lib, album=self.album)
 
-    def emit_created(self, session):
-        """Send the `import_task_created` event for this task.
+    def handle_created(self, session):
+        """Send the `import_task_created` event for this task. Return a list of
+        tasks that should continue through the pipeline. By default, this is a
+        list containing only the task itself, but plugins can replace the task
+        with new ones.
         """
-        plugins.send('import_task_created', session=session, task=self)
+        tasks = plugins.send('import_task_created', session=session, task=self)
+        if not tasks:
+            tasks = [self]
+        else:
+            # The plugins gave us a list of lists of task. Flatten it.
+            flat_tasks = []
+            for inner in tasks:
+                flat_tasks += inner
+            tasks = [t for t in flat_tasks if t]
```

Comment:
> Are we expecting some of the tasks to be `None`?

No, not realy. `None` as a task is not expected nor useful.
> Is this step necessary?

Good question. It depends on how defensive we want it. It was the same with the check for list or single object: When the plugin developer returns a wrong object (or `None` in this case), beets will crash. Since we removed the check for a list we should also remove this check for `None`.

`[t for inner in tasks for t in inner]` :+1: